tests/create_guardduty_member/main.tf

```diff
@@ -12,22 +12,22 @@ provider "aws" {
 }
 
 # Create GuardDuty detector for the administrator account
-resource "aws_guardduty_detector" "administrator" {
-  provider = aws.administrator
-  enable   = true
+module "guarduty" {
```

Comment:
This isn't quite right. You need to leave this as it was. And then _add_ a new module block that invokes the top-level module, which will manage the guardduty detector in the member account.